static/app/views/alerts/rules/details/metricHistory.tsx

```diff
@@ -0,0 +1,244 @@
+import * as React from 'react';
+import {css} from '@emotion/react';
+import styled from '@emotion/styled';
+import moment from 'moment-timezone';
+
+import CollapsePanel from 'sentry/components/collapsePanel';
+import DateTime from 'sentry/components/dateTime';
+import Duration from 'sentry/components/duration';
+import ErrorBoundary from 'sentry/components/errorBoundary';
+import Link from 'sentry/components/links/link';
+import PanelTable from 'sentry/components/panels/panelTable';
+import Tooltip from 'sentry/components/tooltip';
+import {IconShow} from 'sentry/icons';
+import {t, tct, tn} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {Organization} from 'sentry/types';
+import {AlertRuleThresholdType} from 'sentry/views/alerts/incidentRules/types';
+import {Incident, IncidentActivityType, IncidentStatus} from 'sentry/views/alerts/types';
+import {alertDetailsLink} from 'sentry/views/alerts/utils';
+import {AlertWizardAlertNames} from 'sentry/views/alerts/wizard/options';
+import {getAlertTypeFromAggregateDataset} from 'sentry/views/alerts/wizard/utils';
+
+const COLLAPSE_COUNT = 3;
+
+export function getTriggerName(value: string | null) {
+  if (value === `${IncidentStatus.WARNING}`) {
+    return t('Warning');
+  }
+
+  if (value === `${IncidentStatus.CRITICAL}`) {
+    return t('Critical');
+  }
+
+  // Otherwise, activity type is not status change
+  return '';
+}
+
+type Props = {
+  organization: Organization;
+  incidents?: Incident[];
+};
+
+function MetricHistory({organization, incidents}: Props) {
+  const renderActivity = incident => {
```

Comment:
because `renderActivity` is a bit big i'd prefer if it was its own function/component. Can be in the same file and just pass the org + incident as props.